auto-diagram/README.md

```diff
@@ -0,0 +1,66 @@
+# auto-diagram
+Creating a layer of automation for diagram creation
+
+## API Design
+
+The API for this application is documented in the `spec` directory. We are tentatively planning to provide a REST API for creating diagrams. Ansible could send requests to this API when creating a new environment. Alternately, we might choose to treat this as a build step. In that case, we'd keep the data structure defined in `spec` but discard the REST API endpoints.
+
+The data would come from a combination of Ansible inventories (e.g., the environment names) and data provided by the engagement lead (e.g. the company name).
+
+In the future, multiple tools could potentially consume the same data. For example, we might generate documentation as well as diagrams.
+
+We're following the philosophy laid out in [Consumer-Driven Contracts](https://martinfowler.com/articles/consumerDrivenContracts.html) to ensure that our API is meaningful and extensible. In particular:
+
+- Provide just the data that the consumer (the diagram generator) needs, not all the info that Ansible has to offer.
+- Expect the consumer to ignore new fields that it doesn't understand. That way, we could update the API without breaking consumers that aren't able to update right away.
+
+The API is versioned using [Semver](https://semver.org/). Backwards-compatible feature updates (e.g., new properties added) are indicated by minor releases. Breaking changes (e.g., an endpoint is renamed or a property is removed) are indicated by major releases.
+
+## Development
+### Using a Container
+#### Previewing an image
+1. Clone the repository.
+2. Navigate to the `auto-diagram` of the repository.
+3. Review the `/JSONS/` folder to see the different pre-set json files or add your own custom json file there and copy it to the `public/example-api.json`, overwriting the existing file.
+4. Run `$ docker image build .` to build the docker image.
+5. Run `$ docker run -it <IMAGE ID> bash`.
+ - Run `$ docker images` to reveal the IDs of all images.
+6. Run `# npm run serve` to run the npm server inside of the container.
+7. Navigate to the Network URL in your favorite browser (preferably chrome) to preview the image.
+- any modifications you make to the `./public/example-api.json` file inside of the container will be reflected in this preview
+
+#### Taking a screenshot
+0. Follow steps 1-5 from **Previewing an image**.
+1. Run `$ ./container.sh` to build the image inside of your built.
+2. In a separate terminal, run `$ docker cp <CONTAINER ID>:/app/output.jpeg /screenshot/local/path` to copy the image from the container
```

Comment:
It might be worth adding a `docker container ls` step to get the container ID. The first time I went through these instructions, I read through this quickly and tried with the image ID, which of course didn't work.